modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/TableRowConverterImpl.java

```diff
@@ -56,15 +74,41 @@ public class TableRowConverterImpl implements TableRowConverter {
 
         int size = requiredColumns == null ? descriptor.columnsCount() : requiredColumns.cardinality();
 
-        mapping = new int[size];
+        requiredColumnsMapping = new int[size];
+        fullMapping = new int[descriptor.columnsCount()];
+        keyMapping = new int[schemaDescriptor.keyColumns().length()];
+
+        for (int i = 0, j = 0, k = 0; i < descriptor.columnsCount(); i++) {
+            ColumnDescriptor column = descriptor.columnDescriptor(i);
+            int schemaIndex = schemaDescriptor.column(column.name()).schemaIndex();
```

Comment:
this looks overcomplicated a bit... You actually need only schema descriptor to build this converter. To do this, you need to simple iterate over columns in the order they appear in schema descriptor, and use `org.apache.ignite.internal.schema.Column#columnOrder` to get index of the column from sql row (i.e. `fullMapping[i] = schemaDescriptor.column(i).columnOrder()`)

Reply:
Fixed. Removed a dependency on TableDescriptor from TableRowConveterImpl.